File: longest-repeating-character-replacement.cc
// Longest Repeating Character Replacement
class Solution {
public:
  int characterReplacement(string s, int k) {
    int x = 0, mx = 0, c[127] = {};
    for (int i = 0; i < s.size(); i++) {
      mx = max(mx, ++c[s[i]]);
      if (mx+k <= i-x)
        c[s[x++]]--;
    }
    return s.size()-x;
  }
};

File: longest-repeating-character-replacement.nim
# Longest Repeating Character Replacement
proc characterReplacement(s: cstring, k: int): int {.exportc.} =
  var
    l = 0
    r = 0
    mx = 0
    c: array[127, int]
  while s[r] != '\0':
    inc c[s[r].ord]
    mx = mx.max(c[s[r].ord])
    inc r
    if mx+k < r-l:
      dec c[s[l].ord]
      inc l
  r-l
